### ISSUE TITLE
More restrictive block-id character set in Doxygen >= 1.10

### ISSUE THREAD

**Issue by coryan** [doxygen/doxygen#10859]
**Describe the bug**

Doxygen 1.9 supported

```
@snippet{doc} foo.cc bar-baz
```

With Doxygen 1.10 the `-baz` portion is ignored and the snippet cannot be found.

**To Reproduce**

[doxygen-repro.tar.gz](https://github.com/doxygen/doxygen/files/15266137/doxygen-repro.tar.gz)

Extract the tarball and run:

```
doxygen
```

The Dockerfile may be convenient if you don't have Doxygen 1.9 and 1.10
available. Fedora:39 ships with Doxygen 1.9, and this passes:

```
docker build --build-arg=VERSION=39 -f Dockerfile . -t doxy:39
```

while this fails:

```
docker build --build-arg=VERSION=40 -f Dockerfile . -t doxy:40
```

**Expected behavior**

I expected the repro to succeed with Doxygen >= 1.10.

**Version**

```
doxygen --version
1.10.0 (GIT-NOTFOUND)
```

**Additional context**

I think the change in behavior was introduced as part of:

https://github.com/doxygen/doxygen/commit/7780d514980a97cfc2a5f21761ed7da338466350

Specifically this:

https://github.com/doxygen/doxygen/blob/3ce42bf08db82a4fb023a4310462557640c5262d/src/commentcnv.l#L225

The regular expression for `ID` does not accept spaces, or dashes, or a number of other characters. If this is intentional, documenting the accepted character set would be useful.

**Comment by albert-github** [comment]
In the doxygen 1.9.8 version we see that the relevant code (in this case it is still in doctokenizer.l) is:
```
<St_Snippet>[^\\\n]+   {
                         yyextra->token.name += yytext;
                       }
<St_Snippet>"\\"       {
                         yyextra->token.name += yytext;
                       }
<St_Snippet>(\n|"\\ilinebr")  {
                         unput_string(yytext,yyleng);
                         yyextra->token.name = yyextra->token.name.stripWhiteSpace();
                         return TK_WORD;
                       }
```
in other words every character except a line ending was allowed.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10862

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
